pytorch_lightning/loops/dataloader/evaluation_dataloader_loop.py

```diff
@@ -99,6 +94,14 @@ def reset(self) -> None:
         if isinstance(self._max_batches, int):
             self._max_batches = [self._max_batches] * len(self.dataloaders)
 
+    def on_skip(self):
+        return []
+
+    def run(self, *args: Any, **kwargs: Any) -> Optional[Any]:
+        assert self.trainer.evaluating or self.trainer.sanity_checking
+        # the model is set to eval mode in `on_run_start()` and back to train mode in `on_run_end()`
+        return super().run(*args, **kwargs)
```

Comment:
can we just remove it? 
We also dont have assert trainer.training in fit loop etc
cc @carmocca

Reply:
We can now but could be helpful when we allow people to call run at free will

Reply:
```suggestion
```